### PR TITLE
externalUserIds added and data format changed as object

### PR DESCRIPTION
**Pull request by karayel**
According to documentation, data is an object and there are some new variables .
I just added include_external_user_ids 

Please, check documentation for more details.
https://documentation.onesignal.com/reference#create-notification

### CODE REVIEW

File: src/main/java/com/currencyfair/onesignal/model/notification/Notification.java [hunk 1]
```diff
@@ -193,7 +194,7 @@
      * <p>
      * Example: {@code {"abc": "123", "foo": "bar"}}
      */
-    private Map<String, String> data;
+    private JsonNode data;
```
Comment (sriprasanna) [inline src/main/java/com/currencyfair/onesignal/model/notification/Notification.java:197]: Any specific reason to switch from Map to JsonNode?

**Review by sriprasanna — COMMENTED** [review]
One comment otherwise looks good to me.

File: src/main/java/com/currencyfair/onesignal/model/notification/Notification.java (hunk 1 above)
Comment (karayel) [reply src/main/java/com/currencyfair/onesignal/model/notification/Notification.java:197]: There are two main reasons to do that. First of all, I just want to pass value different than String like Object, int, boolean etc. Secondy, I am trying to send complex json via data and it is hard to create it with using map and also according to documentation, data is object so it seems to me JsonNode is better option but at that moment, we break the backward compatibility. I am open to suggestion and I can update pull request as your recommendation.

Comment (yakuninv) [reply src/main/java/com/currencyfair/onesignal/model/notification/Notification.java:197]: I agree that Map<String, String> significantly limits functionality. Usage of Map<String, Object> may minimize impact. But it is still not fully backwards compatible.

Comment (sriprasanna) [reply src/main/java/com/currencyfair/onesignal/model/notification/Notification.java:197]: Fair enough. Some of the code has changed, @karayel if you can pull the latest changes and up the code, I am happy to merge the changes.

Comment (sriprasanna) [reply src/main/java/com/currencyfair/onesignal/model/notification/Notification.java:197]: BTW, the `externalUserIds` is already available https://github.com/CurrencyFair/OneSignal-Java-SDK/blob/master/src/main/java/com/currencyfair/onesignal/model/notification/Notification.java#L101

**Pull request closed without merging**